libraries/kiwi/kiwi-triplestore/src/main/java/org/apache/marmotta/kiwi/sail/KiWiValueFactory.java

```diff
@@ -565,6 +565,8 @@ public Statement createStatement(Resource subject, URI predicate, Value object,
 
             KiWiTriple result = new KiWiTriple(ksubject,kpredicate,kobject,kcontext);
 
+            boolean needsDBLookup = false;
+
             synchronized (registry) {
```

Comment:
I wondered about that too, but thought I'd have to change the `KiWiTripleRegistry` interface and add the lock to the three implementations.  Wasn't sure if it was worth the reshuffling :)